db/migrate/20220505191131_add_last_seen_reviewable_id_to_user.rb

```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class AddLastSeenReviewableIdToUser < ActiveRecord::Migration[7.0]
+  def change
+    add_column :users, :last_seen_reviewable_id, :integer, default: 0, null: false
```

Comment:
```suggestion
    add_column :users, :last_seen_reviewable_id, :integer, null: false
```

I'm guessing this is a foreign_key? If that is the case I don't think a `default` of `0` will be useful. Do we know if we'll be filtering against this column and hence might require a foreign key?

Reply:
Hmm I'm not sure I understand. The column will be used to show the user the count of reviewables they have not seen, the query for that would look something like this:

```sql
SELECT COUNT(*) FROM reviewables WHERE id > :last_seen_reviewable_id AND ...more conditions
```
where the `:last_seen_reviewable_id` placeholder will be replaced with `current_user.last_seen_reviewable_id`. This is the same pattern that we use for notifications with the `seen_notification_id` column on users.

Also we have to provide a default if the column isn't nullable.

Reply:
O sorry I meant if we require a index on this column but looking at your use case I don't think we need it. I guess I'm wondering why the column should not be nullable.

Reply:
I feel like it's slightly more "hygienic" to make it not nullable. For example, in the use case I described above if the column was null we wouldn't be able to simply do `Reviewable.where("id > ?", current_user.last_seen_reviewable_id ).count` because if `last_seen_reviewable_id` was null then the query would always return 0 when we'd expect to return the count of all reviewables. To fix that would have to fallback to zero with `|| 0` or `.to_i`.

Reply:
I have a different opinion because I think having the column nullable is fine. In this case, having `0` as the `last_seen_reviewable_id` doesn't feel right because there is no `reviewable` with an `id` of `0`. We're kind of letting how we intend to query against the database decide how the data should be modelled. 

In the code base, I would simply check if there is actually a `last_seen_reviewable_id` set and then alter the query filter based on that conditional.

Reply:
Alright, updated the migration!